api/v1beta1/gcpmachine_types.go

```diff
@@ -89,6 +124,22 @@ type GCPMachineSpec struct {
 	// +optional
 	Image *string `json:"image,omitempty"`
 
+	// AcceleratorConfigs is the gpuAccelerator configuration for the GCP machine.
+	// +optional
+	AcceleratorConfigs *AcceleratorConfig `json:"acceleratorConfigs,omitempty"`
+
+	// OnHostMaintenance is the action to take when the host machine is being upgraded.
+	// It is either "TERMINATE" or "MIGRATE" depending on the machine type and preemptibility.
+	// +kubebuilder:validation:Enum=TERMINATE;MIGRATE
+	// +kubebuilder:default=MIGRATE
+	// +optional
+	OnHostMaintenance string `json:"onHostMaintenance,omitempty"`
```

Comment:
As you've created a typed string, this field should be using that typed string
```suggestion
	OnHostMaintenance OnHostMaintenance `json:"onHostMaintenance,omitempty"`
```

Also, and I'm not sure how much this project will agree, Kube API conventions would recommend that the constant values here are `Terminate` and `Migrate`, matching the PascalCase convention for enums, rather than matching the cloud provider values

Not sure what the maintainers prefer but thought I should mention

Reply:
Thanks for raising this @JoelSpeed...especially considering the conversation on the confidential compute PR.

Which reminds me i need to get the image-builder PR fixed/merged....